packages/react-search-ui-views/src/view-helpers/screenReaderAnnouncements.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * This helper creates a live region that announces the results of certain
+ * actions (e.g. searching, paging, etc.), that are otherwise invisible
+ * to screen reader users.
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions
+ */
+const regionId = "search-ui-screen-reader-announcements";
+const hasDOM = typeof document !== "undefined"; // Prevent errors in SSR apps
+
+const getLiveRegion = () => {
+  let region = document.getElementById(regionId);
+  if (region) return region;
+
+  region = document.createElement("div");
+  region.id = regionId;
+  region.setAttribute("role", "status");
+  region.setAttribute("aria-live", "polite");
+
+  /**
+   * Visually-hidden CSS that's still available to screen readers.
+   * We're avoiding putting this in a stylesheet to ensure that this
+   * still works for users that opt for custom CSS instead of importing
+   * Search UI's styles. We're also opting to use CSSOM instead of
+   * inline styles to avoid Content Security Policy warnings.
+   *
+   * @see https://github.com/h5bp/html5-boilerplate/blob/v5.0.0/src/css/main.css#L126-L140
+   */
+  region.style.position = "absolute";
+  region.style.width = "1px";
+  region.style.height = "1px";
+  region.style.margin = "-1px";
+  region.style.padding = "0";
+  region.style.border = "0";
+  region.style.overflow = "hidden";
+  region.style.clip = "rect(0 0 0 0)";
+
+  document.body.appendChild(region);
+  return region;
+};
+
+const announceToScreenReader = announcement => {
+  if (hasDOM) {
```

Comment:
Do we need to do explicit DOM manipulation? Could this a more declarative, React type thing? Like could announcements be managed in central state somewhere? 

For instance, messages could go into their own context store and we could just wire a ScreenReaderStatus component up to that context and re-render when it changes.

We could even take it a step further and make screen reader status part of the core Search UI state, and have an action to update it.

Reply:
If it were part of the core Search UI state, it would make it available for any library also, whether it's Vue, React, or vanilla js, etc.

Reply:
At some point we'll probably want to do explicit DOM manipulation (we could use a React portal, but that's also essentially just DOM appending/manipulation and will still need a document/window check for SSR apps). Also FWIW, I looked at [how downshift was doing announcements](https://github.com/downshift-js/downshift/blob/master/src/set-a11y-status.js) as well after noticing their live region, and they're essentially doing what we're doing, which gives me a bit of confidence.

Reply:
Ah, wait, no, I think I maybe see what you mean by central state. Originally when I was making this, I was really only thinking in terms of putting this in `react-search-ui-views`, and my hesitation towards putting it in, e.g. `<Layout>` was something along the lines of "What if the developer isn't using our `Layout` component".

But if we're moving to `react-search-ui` instead (correct me if I'm wrong), then we can take advantage of the top level `<SearchProvider>` component itself to create the region + manage a central state.

Is that what you were thinking?

Reply:
Or, wait, sorry, I'm still catching up again :) You're thinking even higher level than that - putting it in [SearchDriver.js](https://github.com/elastic/search-ui/blob/master/packages/search-ui/src/SearchDriver.js) (or it's own file?) - correct me if I'm wrong?

I'd love to get a chance to chat with you more about this to clarify architecture a bit more!

Reply:
@constancecchen Yes, I think this warrants a chat. Will message you later today!

Reply:
Summary of our chat: https://github.com/elastic/search-ui/issues/325